### ISSUE TITLE
Containers report one extra CPU core

### ISSUE THREAD

**Issue by Sangeppato** [docker/for-mac#7010]
### Description

In Docker Desktop 4.24 the containers see one extra CPU core compared to the ones actually available to the VM

### Reproduce

1. Check the number of CPU cores allocated in the Resources settings
2. Check the actual number of cores seen by any container running `lscpu` (or even in the "Containers" view in the Dashboard)

### Expected behavior

The correct number of CPU cores should be seen

### docker version

```bash
Client:
 Cloud integration: v1.0.35+desktop.5
 Version:           24.0.6
 API version:       1.43
 Go version:        go1.20.7
 Git commit:        ed223bc
 Built:             Mon Sep  4 12:28:49 2023
 OS/Arch:           darwin/arm64
 Context:           desktop-linux

Server: Docker Desktop 4.24.0 (122432)
 Engine:
  Version:          24.0.6
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.7
  Git commit:       1a79695
  Built:            Mon Sep  4 12:31:36 2023
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.22
  GitCommit:        8165feabfdfe38c65b599c4993d227328c231fca
 runc:
  Version:          1.1.8
  GitCommit:        v1.1.8-0-g82f18fe
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    24.0.6
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.5
    Path:     /Users/lll/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.22.0-desktop.2
    Path:     /Users/lll/.docker/cli-plugins/docker-compose
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /Users/lll/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     /Users/lll/.docker/cli-plugins/docker-extension
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.8
    Path:     /Users/lll/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/lll/.docker/cli-plugins/docker-sbom
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     /Users/lll/.docker/cli-plugins/docker-scan
  scout: Docker Scout (Docker Inc.)
    Version:  v1.0.7
    Path:     /Users/lll/.docker/cli-plugins/docker-scout

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 1
 Server Version: 24.0.6
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8165feabfdfe38c65b599c4993d227328c231fca
 runc version: v1.1.8-0-g82f18fe
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 6.4.16-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 4
 Total Memory: 7.765GiB
 Name: docker-desktop
 ID: dcb3c291-6f9d-4490-8cdf-c12a999bc5b2
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

7534C0A5-9659-4A6B-A7E3-299BBD4B9F2C/20231003151833

### Additional Info

- Resources settings:
<img width="494" alt="Screenshot 2023-10-03 at 17 12 12" src="https://github.com/docker/for-mac/assets/37263823/2833a2fe-026c-49fa-9dfc-4a21da0e8527">

- Containers view:
<img width="249" alt="Screenshot 2023-10-03 at 17 12 34" src="https://github.com/docker/for-mac/assets/37263823/7c300d0e-d4e1-4ea3-89f7-4595540682a9">

EDIT:

- When running `stress` inside a container I can see that the VM utilizes the correct number of cores.

- Inside a container `nproc` reports the correct number of cores, while `nproc --all` `lscpu` and `cat /proc/cpuinfo` all report one extra core. So it looks like the VM has access to one extra core but each process is limited to the correct number (as indicated by `nproc`)?

**Comment by taxoakumi2810** [comment]
Hello.

I have installed CPU resources of 56 cores, but containersview only has 10 cores allocated.
I do not understand why?

![image](https://github.com/docker/for-mac/assets/147679314/087b077c-889b-4756-a093-fb345b856d42)

**Comment by Sangeppato** [comment]
Looks like it's fixed in 4.25.0
